kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/k8s/HelmClient.java

```diff
@@ -152,4 +162,38 @@ private String mapToValuesParameterArgument(Map<String, String> m) {
     private static String concatEntry(Map.Entry<String, String> entry) {
         return String.format("%s=%s", entry.getKey(), entry.getValue());
     }
+
+    /**
+     * Gets chart version.
+     *
+     * @param chartName the chart name
+     * @param releaseVersion the release version
+     * @return the chart versions
+     */
+    public String getChartVersion(String chartName, String releaseVersion) {
+        String chartVersion = "";
+        if (releaseVersion != null && !releaseVersion.isEmpty()) {
+            LOGGER.info("Getting chart version for {}", chartName);
+            ExecResult result = Exec.exec(null, command("search", "repo",
+                    chartName,
+                    "--versions",
+                    "--output=json"), 0, true);
+
+            ObjectMapper mapper = new ObjectMapper();
+            try {
+                ArrayNode rootNode = (ArrayNode) mapper.readTree(result.out());
+                // Looking for the latest version that match with releaseVersion. If set to 1.15, it will look for the latest 1.15.Z
+                for (JsonNode node : rootNode) {
+                    if (node.at("/app_version").asText().startsWith(releaseVersion)) {
```

Comment:
That's not safe.  A release version of `1.1` would match 1.19!  You could parse the parts at integers and match those, or make sure a you search string end in dot.  If you do the latter, you need to be sure that the input is just `<major>.<minor>`.

Reply:
I agree with your motivation that we should look for the latest patch release, but as @k-wall says we need to be a bit more sophisticated about how we do it. 

An other alliterative  you could try is `releaseVersion.split(".")` and `node.at("/app_version").asText().split(".")` and compare the first two array elements to make sure you have found a match. 

That assumes you still only get one result per `major.minor` pair, if not you have to keep iterating to find the highest patch number.